test/helpers/votingTestCases.js

```diff
@@ -0,0 +1,188 @@
+import { expect } from "chai";
+import { increaseTime } from "../helpers/evmCommands";
+import { createSignedVote } from "../helpers/relayedVoteSigning";
+import { prettyPrintGasCost } from "../helpers/gasUtils";
+
+export function testVotingWithSnapshots(getToken, getVotingContract, owner, owner2, relayer) {
```

Comment:
yeah big 👍 for test coverage. actually we use form of test suites that can be used against any contract just for the standard interfaces (like ERC20. Snapshot, TokenController etc.) but it's pretty cool that you did it this way, following this pattern.